aws/src/main/java/org/apache/iceberg/aws/glue/GlueCatalog.java

```diff
@@ -162,7 +165,21 @@ private String cleanWarehousePath(String path) {
 
   @Override
   protected TableOperations newTableOps(TableIdentifier tableIdentifier) {
-    return new GlueTableOperations(glue, lockManager, catalogName, awsProperties, fileIO, tableIdentifier);
+    if (awsProperties.glueLakeFormationEnabled()) {
```

Comment:
if we only assign `catalogProperties` when necessary, it means we don't need to check for `if(awsProperties.glueLakeFormationEnabled())` here, but just `if(catalogProperties != null)`